src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -3983,5 +3984,62 @@ public void AssertNotDisposed()
             if (_disposeOne.Disposed)
                 ThrowObjectDisposed();
         }
+
+        public int Dump(string path, Action<IOperationProgress> onProgress)
+        {
+            if (Directory.Exists(path) == false)
+                Directory.CreateDirectory(path);
+
+            using (var tx = _environment.ReadTransaction())
+            {
+                var state = new Indexing.VoronState(tx);
+
+                var files = IndexPersistence.LuceneDirectory.ListAll(state);
+                var currentFile = 0;
+                var buffer = new byte[64 * 1024];
+                var sp = Stopwatch.StartNew();
+                foreach (var file in files)
+                {
+
+                    using (var input = IndexPersistence.LuceneDirectory.OpenInput(file, state))
+                    using (var output = File.Create(Path.Combine(path, file)))
+                    {
+                        var len = input.Length(state);
+                        string message = "Exporting file: " + file;
+                        onProgress(new AdminIndexHandler.DumpIndexProgress
+                        {
+                            Message = message,
+                            TotalFiles = files.Length,
+                            ProcessedFiles = currentFile,
+                            TotalBytes = len
```

Comment:
This doesn't look like TotalBytes? It is per file?

Reply:
That is the total bytes for _this_ file, yes. Need to change this